src/main/java/org/myrobotlab/framework/repo/IvyWrapper.java

```diff
@@ -389,7 +389,12 @@ synchronized public void install(String location, String[] serviceTypes) {
       Set<ServiceDependency> targetLibraries = getUnfulfilledDependencies(serviceTypes);
 
       if (targetLibraries.size() == 0) {
-        info("%s already installed", (Object[]) serviceTypes);
+        StringBuilder sb = new StringBuilder();
+        for (String type : serviceTypes) {
+          sb.append(type.substring(type.lastIndexOf(".") + 1));

```

Comment:
This logic is fragile as it could yield an out of bounds array problem.

Reply:
fixed